extensions/amp-consent/0.1/consent-info.js

```diff
@@ -328,27 +399,53 @@ export function getConsentStateValue(enumState) {
 }
 
 /**
- * Converts metadata to stroage value:
- * {'gdprApplies': true, 'consentStringType': 'tcf-v2'} =>
- * {'ga': true, 'cst': 'tcf-v2'}
+ * Convert the ConsentMetadataDef back to readable object
+ * @param {ConsentMetadataDef=} metadata
+ * @return {Object|undefined}
+ */
+export function getConsentMetadataValue(metadata) {
+  if (!metadata) {
+    return undefined;
+  }
+  const metadataValue = {};
+  if (metadata['consentStringType']) {
+    metadataValue['consentStringType'] = convertConsentStringType(
+      metadata['consentStringType']
+    );
+  }
+  return metadataValue;
+}
+
+/**
+ * Converts ConsentMetadataDef to stroage value:
+ * {'gdprApplies': true, 'consentStringType': 2} =>
+ * {'ga': true, 'cst': 2}
  *
- * @param {Object} unused
+ * @param {ConsentMetadataDef=} consentInfoMetadata
  * @return {Object}
  */
-function composeMetadataStoreValue(unused) {
+export function composeMetadataStoreValue(consentInfoMetadata) {
   const storageMetadata = map();
-  // TODO(micajuineho) if (metadata['gdprApplies']) {...}
+  if (consentInfoMetadata['consentStringType']) {
+    storageMetadata[METADATA_STORAGE_KEY.CONSENT_STRING_TYPE] =
+      consentInfoMetadata['consentStringType'];
+  }
   return storageMetadata;
 }
 
 /**
- * TODO(micajuineho) Converts stroage metadata to human readable object:
- * {'ga': true, 'cst': 'tcf-v2'} =>
- * {'gdprApplies': true, 'consentStringType': 'tcf-v2'}
+ * TODO(micajuineho) Converts stroage metadata to ConsentMetadataDef:
+ * {'ga': true, 'cst': 2} =>
+ * {'gdprApplies': true, 'consentStringType': 2}
  *
- * @param {Object|null|undefined} unused
- * @return {undefined}
+ * @param {Object|null|undefined} storageMetadata
+ * @return {ConsentMetadataDef|undefined}
  */
-function convertStorageMetadata(unused) {
-  return undefined;
+export function convertStorageMetadata(storageMetadata) {
+  if (!storageMetadata) {
+    return undefined;
```

Comment:
I would avoid `undefined` in general if there's no special reason. `undefined` is more error prone than `null`.

Reply:
We should return undefined here because it's getting used in `constructConsentInfo`, where the standard is to have the value or undefined for each of its fields.

Reply:
I see. Ok